src/coreclr/gcinfo/gcinfohelpers.h

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#ifndef _GCINFOHELPERS_H_
+#define _GCINFOHELPERS_H_
+
+#undef _ASSERTE
+
+#if defined(_DEBUG)
+extern "C" void __cdecl assertAbort(const char* why, const char* file, unsigned line);
```

Comment:
Putting it here is the reason why you needed to add the assertAbort for DAC. This file is included by the gcinfodumper.cpp too and that one is used by DAC only. If you had it defined for gcinfoencoder.cpp and arraylist.cpp only, it would not require that. Please note that gcinfodumper.coo is not part of the gcinfo static library, it is directly included by the DAC build.